cpp/benchmarks/copying/contiguous_split_benchmark.cu

```diff
@@ -70,18 +73,26 @@ void BM_contiguous_split(benchmark::State& state)
   int64_t num_rows        = total_desired_bytes / (num_cols * el_size);
 
   // generate input table
-  srand(31337);
-  auto valids = cudf::detail::make_counting_transform_iterator(0, [](auto i) { return true; });
-  std::vector<cudf::test::fixed_width_column_wrapper<int>> src_cols(num_cols);
+  auto valids = cudf::detail::make_counting_transform_iterator(
+    0, [] __device__(cudf::size_type i) { return true; });
+  std::vector<std::unique_ptr<cudf::column>> src_cols;
+  src_cols.reserve(num_cols);
```

Comment:
I think this can be one line, using the constructor to specify the size.
```suggestion
  std::vector<std::unique_ptr<cudf::column>> src_cols(num_cols);
```

Reply:
This is kind of grey area.  Granted this is concise, This suggestion will call default column constructor `num_cols` times, which in my opinion unnecessary and not useful for the code.
That's why I prefer using `reserve` and `back_inserter`.

Reply:
I don’t have strong feelings about this, either pattern is okay with me. It is a tradeoff of conciseness for a few constructor calls. But this should only construct unique pointers to nullptr and _not_ actual column objects, right?

Reply:
AFAIK, it will build actual column objects by calling default constructor `column()` and allocate memory for its members and call their default constructor as well. It does not incur much cost. But just not useful in any way.
IIRC, `rmm::device_uvector` is created to avoid this specifically.

Quoting  `premature optimization is the root of all evil`,
So, I wouldn't mind changing this for conciseness. 👍

Reply:
The example code here shows that the constructor of `T` is not called when constructing a `unique_ptr<T>` with no arguments or called with `nullptr` as the argument: https://en.cppreference.com/w/cpp/memory/unique_ptr/unique_ptr

Reply:
My bad. That's right. it will be just `unique_ptr<T>()` called.